abi.html

```diff
@@ -5014,6 +5014,10 @@ <h5><a href="#mangling-builtin">5.1.5.2 Builtin types</a></h5>
                  ::= Df # IEEE 754r decimal floating point (32 bits)
                  ::= Dh # IEEE 754r half-precision floating point (16 bits)
                  ::= DF &lt;<a href="#mangle.number">number</a>&gt; _ # ISO/IEC TS 18661 binary floating point type _FloatN (N bits)
+                 ::= DB &lt;<a href="#mangle.number">number</a>&gt; _        # signed _BitInt(N)
+                 ::= DB &lt;<a href="#mangle.template-args">template-args</a>&gt; _ # signed _BitInt(N)
+                 ::= DU &lt;<a href="#mangle.number">number</a>&gt; _        # unsigned _BitInt(N)
+                 ::= DU &lt;<a href="#mangle.template-args">template-args</a>&gt; _ # unsigned _BitInt(N)
```

Comment:
`<number> | <expression>` is an unambiguous grammar (it's what we use for array types).  If we really need extra arguments in the future, `<number> | <expression> | <template-args>` will still be unambiguous.

Please cite C23 in the comments.

Reply:
Heh, I was using `|` originally but decided to remove it because I couldn't find a use of the character within abi.html. FWIW, this is what I see for [array types](http://itanium-cxx-abi.github.io/cxx-abi/abi.html#mangle.array-type):
```
  <array-type> ::= A <positive dimension number> _ <element type>
	       ::= A [<dimension expression>] _ <element type>
```
I made the switch anyway because I agree that `|` should be sufficiently clear, but if you want me to switch back, I can do so easily enough.

Reply:
Uh, sorry, I was using that as shorthand.  It's best to stay consistent with the rest of the document, which always uses separate lines for alternative productions.  My point was just meant to be about using `<expression>` as the sub-production rather than `<template-args>`.

Reply:
Ah, sorry for getting that wrong, I've corrected it now. Thanks!